pyfarm/agent/sysinfo/network.py

```diff
@@ -93,11 +93,21 @@ def hostname(trust_name_from_ips=True):
     """
     logger.debug("Attempting to discover the hostname")
 
+    ip_addresses = addresses()
+
+    local_fqdn_query = socket.getfqdn().lower()
+    _h, _a, ips_from_fqdn = socket.gethostbyname_ex(local_fqdn_query)
```

Comment:
Probably need to wrap `socket.gethostbyname_ex` a bit so we ignore the most common case of a name the system still can't resolve (I've seen this happen before on misconfigured systems):

```python
try:
    _h, _a, ips_from_fqdn = socket.gethostbyname_ex(local_fqdn_query)
except socket.gaierror as error:
    if error.errno != ENOEXEC:
        raise
    ips_from_fqdn = set()
```